ocs_ci/deployment/flexy.py

```diff
@@ -291,45 +270,69 @@ def merge_flexy_env(self):
-        and do this only if its jenkins run
-
+        Prepare Flexy working directory (flexy-dir):
+            - copy flexy-dir from cluster_path to data dir (if available)
+            - set proper ownership
         """
-        abs_cluster_path = os.path.abspath(self.cluster_path)
-        flexy_cluster_path = os.path.join(
-            self.flexy_host_dir, "flexy/workdir/install-dir"
-        )
-        if os.path.exists(abs_cluster_path):
-            os.rmdir(abs_cluster_path)
-        # Check whether its a jenkins run
-        if self.is_jenkins_mount():
-            flexy_nfs_path = os.path.join(
-                constants.JENKINS_NFS_CURRENT_CLUSTER_DIR, constants.FLEXY_HOST_DIR
+        logger.info(f"Prepare flexy working directory {self.flexy_host_dir}.")
+        if not os.path.exists(self.flexy_host_dir):
+            # if ocs-ci/data were cleaned up (e.g. on Jenkins) and flexy-dir
+            # exists in cluster dir, copy it to the data directory, othervise
+            # just create empty flexy-dir
+            cluster_path_flexy_dir = os.path.join(
+                self.cluster_path, constants.FLEXY_HOST_DIR
             )
-            if not os.path.exists(flexy_nfs_path):
+            if os.path.exists(cluster_path_flexy_dir):
                 shutil.copytree(
+                    cluster_path_flexy_dir,
                     self.flexy_host_dir,
-                    flexy_nfs_path,
                     symlinks=True,
                     ignore_dangling_symlinks=True,
                 )
-                chmod = f"chmod -R 777 {flexy_nfs_path}"
-                run_cmd(chmod)
-                logger.info(f"Symlinking {abs_cluster_path} to {flexy_nfs_path}")
-                os.symlink(
-                    os.path.join(flexy_nfs_path, constants.FLEXY_RELATIVE_CLUSTER_DIR),
-                    abs_cluster_path,
-                )
-        else:
-            # recursively change permissions
-            # for all the subdirs
-            chmod = f"sudo chmod -R 777 {constants.FLEXY_HOST_DIR_PATH}"
-            run_cmd(chmod)
-            logger.info(f"Symlinking {flexy_cluster_path, abs_cluster_path}")
-            os.symlink(flexy_cluster_path, abs_cluster_path)
+            else:
+                os.mkdir(self.flexy_host_dir)
+        # change the ownership to the uid of user in flexy container
+        chown_cmd = (
+            f"sudo chown -R {constants.FLEXY_USER_LOCAL_UID} {self.flexy_host_dir}"
+        )
+        run_cmd(chown_cmd)
 
+    def flexy_backup_work_dir(self):
+        """
+        Perform copying of flexy-dir to cluster_path.
+        """
+        # change ownership of flexy-dir back to current user
+        chown_cmd = f"sudo chown -R {os.getuid()}:{os.getgid()} {self.flexy_host_dir}"
+        run_cmd(chown_cmd)
+        chmod_cmd = f"sudo chmod -R a+rX {self.flexy_host_dir}"
+        run_cmd(chmod_cmd)
+
+        # dirs_exist_ok parameter for shutil.copytree is available in Python
```

Comment:
do we need this commented part ?

Reply:
I think it will be slightly nicer to use pure python whenever possible, but `shutil.copytree` fails, if (some of) the destination directory exists, so I have to use `rsync` instead. In python 3.8 `copytree` method have additional parameter `dirs_exist_ok`, which if it is set to `True`, it will ignore the existing dirs. But if you think, it will be better to remove this commented code here, I have no problem with that - the `rsync` works well...

Reply:
Reconsidered it and removed the commented part, we don't probably want to change this since the `rsync` way works correctly.